openedx/xblock-drag-and-drop-v2
# feat: add optional explanation field to be displayed on clicking show answer

## Description

This PR:

1. adds an optional `Explanation` text area field in the edit popup of studio. When a text is added here, its displayed to the learner below the xblock when `Show Answer` button is pressed.
![image](https://user-images.githubusercontent.com/3189670/180333591-51e944a5-7349-47e1-b16d-237a1b62dfbe.png)


2. fixes an issue where the feedback div was displayed even when no feedback content was present. This resulted an extra line being displayed because of the top margin of the feedback div.

Before:
![image](https://user-images.githubusercontent.com/36229857/179384715-2f19bb41-a22d-4de1-aa7e-f1bca92217dd.png)

After:
![image](https://user-images.githubusercontent.com/36229857/179384739-a73583d6-a6f2-4655-b038-70a424f799a1.png)


## Supporting information

OpenCraft internal ticket : [BB-6464](https://tasks.opencraft.com/browse/BB-6464)

## Testing instructions

1. Checkout this branch in <devstack_parent>/src directory in your local setup
2. Login to Studio shell, switch to edxapp virtual env
4. Go to /edx/src/xblock-drag-and-drop-v2 and run pip install -e .
5. Go to Studio UI and create a new unit
6. Select Problem > Advanced > Drag and Drop, to create a v2 drag and drop xblock
7. Click Edit and change mode to Assessment
8. Scroll down to `Explanation Text`
9. Add some text in the text area
10. Click Continue twice and Save
11. Click on `View Live Version`
12. Ensure there is no extra line as show above
13. Do as many attempts as required till `Show Answer` button is active
14. Click on `Show Answer`
15. Ensure `Explanation` is displayed below the xblock

## Deadline

"None"
